lib/chef/knife/bootstrap.rb

```diff
@@ -549,7 +549,28 @@ def run
 
         $stdout.sync = true
         register_client
-        connect!
+        begin
+          connect!
+        rescue Train::Transports::SSHFailed => e
+          if e.message =~ /fingerprint (\S+) is unknown for "(.+)"/
+            fingerprint = $1
+            hostname,ip = $2.split(',')
+            puts "The authenticity of host '#{hostname} (#{ip})' can't be established."
+            # TODO: convert the SHA256 base64 value to hex with colons
+            # 'ssh' example output:
+            # RSA key fingerprint is e5:cb:c0:e2:21:3b:12:52:f8:ce:cb:00:24:e2:0c:92.
+            # ECDSA key fingerprint is 5d:67:61:08:a9:d7:01:fd:5e:ae:7e:09:40:ef:c0:3c.
+            puts "fingerprint is #{fingerprint}."
+            ui.confirm("Are you sure you want to continue connecting")
+            # FIXME: this should save the key to known_hosts but doesn't appear to be
+            config[:ssh_verify_host_key] = :accept_new
```

Comment:
train doesn't know about this value - it only expects a bool [1], so we'll need to modify there as well. 

That's probably also why `accept_new` isn't causing `known_hosts` to be updated.  

[1] https://github.com/inspec/train/blob/master/lib/train/transports/ssh.rb#L199

Reply:
That method converts specific values but passes anything else through:

```
2.4.1 :001 > given = :accept_new
 => :accept_new 
2.4.1 :002 >         {
2.4.1 :003 >               'true' => true,
2.4.1 :004 >               'false' => false,
2.4.1 :005 >               nil => false,
2.4.1 :006 >             }.fetch(given, given)
 => :accept_new 
```

Reply:
you can see it working here too, notice after the SNIP in the first and second connections:

```
DEBUG: [SSH] opening connection to ubuntu@ec2-3-85-23-14.compute-1.amazonaws.com<{SNIP, :verify_host_key=>:always}>
The authenticity of host 'ec2-3-85-23-14.compute-1.amazonaws.com (3.85.23.14)' can't be established.
fingerprint is SHA256:dVvJdTZrMcAr+YfN0NV4tnt5Sj5zjdJLhfzEoZvkd90.
Are you sure you want to continue connecting? (Y/N) y

Connecting to ec2-3-85-23-14.compute-1.amazonaws.com
DEBUG: [SSH] Using Agent keys as no password or key file have been specified
DEBUG: [SSH] ubuntu@ec2-3-85-23-14.compute-1.amazonaws.com<{SNIP, :verify_host_key=>:accept_new}> (cmd.exe /c ver)
```

Reply:
Cool, thanks - that's what I get for skimming